internal/server/v2/observation/calculation_util.go

```diff
@@ -245,3 +246,34 @@ func evalExpr(
 		return nil, fmt.Errorf("unsupported ast type %T", t)
 	}
 }
+
+func formatCalculatedResponse(
+	resp *pbv2.ObservationResponse,
+	equation *Equation,
+) error {
+	// Replace placeholder by final variable.
+	variableObs, ok := resp.ByVariable[INTERMEDIATE_NODE]
```

Comment:
It feels like we are using placeholder nodes and manipulating them because we are using `ObservationResponse` as the intermediate object as well.

Not for this PR, but I'm wondering if we can simplify things by using a custom struct for intermediate responses and building `ObservationResponse` only at the end.

Reply:
Good idea, I can do this in a follow up PR and as a precursor to https://github.com/datacommonsorg/mixer/pull/1415